### PR TITLE
Update Content-Type for login request

### PR DESCRIPTION
**Pull request by josunect**
Fixes https://github.com/kiali/kiali/issues/7252

### CODE REVIEW

File: frontend/src/services/Api.ts [hunk 1]
```diff
@@ -102,8 +102,8 @@ const loginHeaders = config.login.headers;
 
 /**  Helpers to Requests */
 
-const getHeaders = (): Partial<AxiosHeaders> => {
-  if (apiProxy) {
+const getHeaders = (urlEncoded?: boolean): Partial<AxiosHeaders> => {
+  if (apiProxy || urlEncoded) {
```
Comment (nrfox) [inline frontend/src/services/Api.ts:106]: It looks like this leaves off `...loginHeaders`. Should those be included for both content types?

Comment (josunect) [reply frontend/src/services/Api.ts:106]: I've thought that loginHeaders contain the login information (Once logged in), but let me verify

Comment (josunect) [reply frontend/src/services/Api.ts]: Updated

Comment (nrfox) [reply frontend/src/services/Api.ts]: I'm wondering if OSSMC should also have those login headers? @ferhoyos?

Comment (ferhoyos) [reply frontend/src/services/Api.ts]: I don't think so, as OSSMC leverages of Openshift login. Where `'X-Auth-Type-Kiali-UI': '1'` is used?

Said that, including loginHeaders probably do not affect OSSMC. Maybe you can simplify the code:
```
    if (apiProxy || urlEncoded) {
      return { 'Content-Type': 'application/x-www-form-urlencoded', ...loginHeaders };
    } else {
      return { 'Content-Type': 'application/json', ...loginHeaders };
   }
```

**Review by nrfox — APPROVED** [review]

**Review by nrfox — APPROVED** [review]

**Pull request merged**